### PR TITLE
BUILD: Add support for `mypy`

### PR DESCRIPTION
**Pull request by mtsokol**

### CODE REVIEW

File: src/sklearndf/classification/wrapper/_wrapper.py [hunk 1]
```diff
@@ -105,27 +115,24 @@ def _prediction_with_class_labels(
...
-            ]
-        else:
-            return [
-                sup._prediction_with_class_labels(X=X, y=output, classes=None)
-                for output in y
-            ]
+        try:
+            estimators = getattr(delegate_estimator, "estimators_")
+        except AttributeError:
+            return [_super._prediction_with_class_labels(X=X, y=output) for output in y]
```
Comment (mtsokol) [inline src/sklearndf/classification/wrapper/_wrapper.py]: I wanted to ask: Is it a Python convention to manage business logic with exceptions? (as I understand the result is the same). I came across same approach in sklearn (https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/_scorer.py#L356) and the issue I had with it is that exception messages are a bit less informative: If an exception happens inside `_prediction_with_class_labels` then in the stack trace first we will have `AttributeError` exception that is expected as this is in our business logic and after that a `while handling x another exception occured` with the actual one. 
Does it make a difference?

Comment (j-ittner) [reply src/sklearndf/classification/wrapper/_wrapper.py]: @mtsokol good point. In general - `hasattr` does the same, catches an `AttributeError` and returns `False` in that case. So the code above avoids a second call to `gestattr`. But the nested exceptions are ugly. So we may want to change the code above to
```python
estimators = getattr(delegate_estimator, "estimators_", None)
if estimators is None:
    …
```

Comment (mtsokol) [reply src/sklearndf/classification/wrapper/_wrapper.py]: Sure! I pushed commit with such changes.

**Review by j-ittner — APPROVED** [review]
Looks great, thanks!

**Pull request merged**